Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -99,34 +121,62 @@ ucum_to_cql_units = {
   , 's':    'second'
   , 'ms':   'millisecond'
 }
-
-get_ucum_unit = (units) ->
-  if ucum_time_units[units]
-    ucum_time_units[units]
-  else units
-
-# The smallest common duration is the millisecond
-# Returns the VALUE of a time duration-based quantity in milliseconds
-durationInMilliseconds = (qty) ->
-  if parseFloat qty.value
-    millivalue = switch
-      when get_ucum_unit(qty.unit) == 's' then qty.value * 1000
-      when get_ucum_unit(qty.unit) == 'min' then qty.value * 60 * 1000
-      when get_ucum_unit(qty.unit) == 'h' then qty.value * 60 * 60 * 1000
-      when get_ucum_unit(qty.unit) == 'd' then qty.value * 24 * 60 * 60 * 1000
-      when get_ucum_unit(qty.unit) == 'wk' then qty.value * 7 * 24 * 60 * 60 * 1000
-      # Support for the UCUM units based on the Gregorian calendar
-      when get_ucum_unit(qty.unit) == 'mo_g' then qty.value * 30.436875 * 24 * 60 * 60 * 1000  # Based on a Gregorian mean month length of 30.436875 days
-      when get_ucum_unit(qty.unit) == 'a_g' then qty.value * 365.2425 * 24 * 60 * 60 * 1000  # Based on a Gregorian year of 365.2425 days
-      # Support for the UCUM units based on the Julian calendar
-      when get_ucum_unit(qty.unit) == 'mo_j' then qty.value * 30.4375 * 24 * 60 * 60 * 1000  # Based on a Julian mean month length of 30.4375 days
-      when get_ucum_unit(qty.unit) == 'a_j' then qty.value * 365.25 * 24 * 60 * 60 * 1000  # Based on a Julian year of 365.25 days
-      else qty.value
-    millivalue
-  else
-    null
-
-module.exports.createQuantity = (value,unit) ->
+# this is used to perform any convertions of CQL date time fileds to their ucum equivalents
+ucum_unit = (unit) ->
+  ucum_time_units[unit] || unit || ''
+
+#just a wrapper function to deal with possible exceptions being thrown
+convert_value = (value, from ,to ) ->
+  try
+    decimalAdjust("round", ucum.convert(value,ucum_unit(from),ucum_unit(to)), -8)
+  catch e
+    throw new IncompatibleTypesException(from, to, e)
```

Comment:
I think they should and I'll fix it.  The ucum library will throw the exception but it's just a string so I felt the need to wrap it a more object oriented fashion.  It also provides a central location for rounding the values.